restore seed page crashes when suggestion doesn't exist

Clicking the return/enter key when a suggestion doesn't exist causes godcr to crash.

### To reproduce
* navigate to restore wallet page
* type in a seed that produces non suggestion
* hit the enter/return key

It should crash with the log


panic: runtime error: index out of range [0] with length 0

goroutine 10 [running]:
github.com/planetdecred/godcr/ui.(*createRestore).handle(0xc0005b0000)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/create_restore_page.go:868 +0x140f
github.com/planetdecred/godcr/ui.(*mainPage).Layout(0xc000305540, 0x0, 0x0, 0x640, 0x4b0, 0x4000000040000000, 0x50fc060, 0xc000774890, 0xc02960f39298ba18, 0x5aad026f9, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/main_page.go:225 +0x102
github.com/planetdecred/godcr/ui.(*Window).layoutPage.func2(0x0, 0x0, 0x640, 0x4b0, 0x4000000040000000, 0x50fc060, 0xc000774890, 0xc02960f39298ba18, 0x5aad026f9, 0x59818a0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/window.go:131 +0x95
gioui.org/layout.Stack.Layout(0x4cc1901, 0x640, 0x4b0, 0x640, 0x4b0, 0x4000000040000000, 0x50fc060, 0xc000774890, 0xc02960f39298ba18, 0x5aad026f9, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/stack.go:60 +0x7d5
github.com/planetdecred/godcr/ui.(*Window).layoutPage(0xc0003d2000, 0x640, 0x4b0, 0x640, 0x4b0, 0x4000000040000000, 0x50fc060, 0xc000774890, 0xc02960f39298ba18, 0x5aad026f9, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/window.go:125 +0x257
github.com/planetdecred/godcr/ui.(*Window).Loop(0xc0003d2000, 0xc000774800, 0xc000531f80)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/window.go:229 +0x409
created by main.main
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/main.go:92 +0x525

--- a/ui/create_restore_page.go
+++ b/ui/create_restore_page.go
@@ -563,30 +563,34 @@ func diff(a, b []int) []int {
 func (pg *createRestore) editorSeedsEventsHandler() {
 	var focused []int
 
+	seedEvent := func(i int, text string) {
+		if pg.seedClicked {
+			pg.seedEditors.focusIndex = -1
+			pg.seedClicked = false
+		} else {
+			pg.seedEditors.focusIndex = i
+		}
+
+		if text == "" {
+			pg.openPopupIndex = -1
+		} else {
+			pg.openPopupIndex = i
+		}
+	}
+
 	for i := 0; i < len(pg.seedEditors.editors); i++ {
 		editor := &pg.seedEditors.editors[i]
+		text := editor.Edit.Editor.Text()
 
 		if editor.Edit.Editor.Focused() {
+			seedEvent(i, text)
 			focused = append(focused, i)
 		}
 
 		for _, e := range editor.Edit.Editor.Events() {
 			switch e.(type) {
 			case widget.ChangeEvent:
-				// hide suggestions if seed clicked
-				if pg.seedClicked {
-					pg.seedEditors.focusIndex = -1
-					pg.seedClicked = false
-				} else {
-					pg.seedEditors.focusIndex = i
-				}
-				text := editor.Edit.Editor.Text()
-				if text == "" {
-					pg.openPopupIndex = -1
-				} else {
-					pg.openPopupIndex = i
-				}
-
+				seedEvent(i, text)
 				pg.resetSeedFields.Color = pg.theme.Color.Primary
 				pg.suggestions = pg.suggestionSeeds(text)
 				pg.seedMenu = pg.seedMenu[:len(pg.suggestions)]
@@ -596,6 +600,7 @@ func (pg *createRestore) editorSeedsEventsHandler() {
 			case widget.SubmitEvent:
 				if i != numberOfSeeds {
 					pg.seedEditors.editors[i+1].Edit.Editor.Focus()
+					pg.selected = 0
 				}
 			}
 		}
@@ -865,6 +870,9 @@ func (pg *createRestore) handle() {
 			}
 		}
 		if (evt.Name == key.NameReturn || evt.Name == key.NameEnter) && pg.openPopupIndex != -1 && evt.State == key.Press && len(pg.suggestions) != 0 {
+			if pg.seedEditors.focusIndex == -1 && len(pg.suggestions) == 1 {
+				return
+			}
 			pg.seedMenu[pg.selected].button.Button.Click()
 		}
 	case err := <-pg.errorReceiver:

--- a/ui/create_restore_page.go
+++ b/ui/create_restore_page.go
@@ -864,7 +864,7 @@ func (pg *createRestore) handle() {
 				pg.selected = len(pg.suggestions) - 1
 			}
 		}
-		if (evt.Name == key.NameReturn || evt.Name == key.NameEnter) && pg.openPopupIndex != -1 && evt.State == key.Press {
+		if (evt.Name == key.NameReturn || evt.Name == key.NameEnter) && pg.openPopupIndex != -1 && evt.State == key.Press && len(pg.suggestions) != 0 {
 			pg.seedMenu[pg.selected].button.Button.Click()
 		}
 	case err := <-pg.errorReceiver:

